Compile error using XCode6.4

Link to travis build error:
https://api.travis-ci.com/v3/job/133798331/log.txt

Here is the specific build error: ```/Users/travis/miniconda3/conda-bld/tiledb_1531257565828/work/tiledb/sm/filesystem/vfs.cc:94:5: error: return type 'const basic_string<[3 * ...]>' must match previous return type 'basic_string<[3 * ...]>' when lambda expression has unspecified explicit return type

    return path;

    ^

/Users/travis/miniconda3/conda-bld/tiledb_1531257565828/work/tiledb/sm/filesystem/vfs.cc:96:5: error: return type 'const basic_string<[3 * ...]>' must match previous return type 'basic_string<[3 * ...]>' when lambda expression has unspecified explicit return type

    return path;

    ^

/Users/travis/miniconda3/conda-bld/tiledb_1531257565828/work/tiledb/sm/filesystem/vfs.cc:98:3: error: return type 'const basic_string<[3 * ...]>' must match previous return type 'basic_string<[3 * ...]>' when lambda expression has unspecified explicit return type

  return path;```

I think it's a compiler bug due to the stats lambda in this function -- we're returning `const std::string&` in one place and `std::string` in another.

--- a/tiledb/sm/filesystem/vfs.cc
+++ b/tiledb/sm/filesystem/vfs.cc
@@ -80,7 +80,8 @@ VFS::VFS() {
 
 std::string VFS::abs_path(const std::string& path) {
   STATS_FUNC_IN(vfs_abs_path);
-
+  // workaround for older clang (llvm 3.5) compilers (issue #828)
+  std::string path_copy = path;
 #ifdef _WIN32
   if (Win::is_win_path(path))
     return Win::uri_from_path(Win::abs_path(path));
@@ -91,11 +92,11 @@ std::string VFS::abs_path(const std::string& path) {
     return Posix::abs_path(path);
 #endif
   if (URI::is_hdfs(path))
-    return path;
+    return path_copy;
   if (URI::is_s3(path))
-    return path;
+    return path_copy;
   // Certainly starts with "<resource>://" other than "file://"
-  return path;
+  return path_copy;
 
   STATS_FUNC_OUT(vfs_abs_path);
 }

